src/briefcase/platforms/macOS/__init__.py

```diff
@@ -202,22 +200,24 @@ def sign_file(self, path, identity, entitlements=None):
             process_command.append(options)
 
         try:
-            self.logger.info(f"Signing {Path(path).relative_to(self.base_path)}")
-            self.subprocess.run(
-                process_command,
-                stderr=subprocess.PIPE,
-                check=True,
-            )
+            with self.input.wait_bar(
+                f"Signing {Path(path).relative_to(self.base_path)}..."
+            ):
+                self.subprocess.run(
+                    process_command,
+                    stderr=subprocess.PIPE,
+                    check=True,
+                )
```

Comment:
This is another example of a "micro-waitbar". Each call to sign should be on the order of microseconds; using a progress bar on each one doesn't really have much of a point.

It *would* be interesting to see if we could use a single waitbar for the entire signing process, with the output of the `subprocess.run()` being used as a sort of "detailed progress". Better still would be to use a *progress* bar, based on the number of files to be signed. Not sure if this will work as well in practice as it does in my head, but it's worth an experiment if you can make this work quickly.

If you want to test out what this would look like in practice - a helloworld app using PySide6 has a lot of signed content, so it's a good testbed.

Reply:
Great idea. I didn't fully grasp this signing process could iterate over potentially many files. I updated to use a progress bar instead. The ETA on the progress bar is a little wonky but the percentage is clearly what matters.